impl/maven-executor/src/main/java/org/apache/maven/cling/executor/embedded/EmbeddedMavenExecutor.java

```diff
@@ -176,7 +176,7 @@ protected void disposeRuntimeCreatedRealms(Context context) {
                 }
             }
         } catch (Exception e) {
-            throw new ExecutorException("Failed to dispose runtime created realms", e);
+            throw new RuntimeException(new ExecutorException("Failed to dispose runtime created realms", e));
```

Comment:
this makes zero sense. First make the actual exception a checked exception, then wrap it into a (untyped) runtime exception.

Now there is no longer a ExecutorException thrown but a RuntimeException.

AI Slop.